src/service/viewport-impl.js

```diff
@@ -1093,8 +1097,11 @@ export class ViewportBindingNatural_ {
 
   /** @override */
   getScrollTop() {
-    return this.getScrollingElement_()./*OK*/scrollTop ||
+    const pageScrollTop = this.getScrollingElement_()./*OK*/scrollTop ||
         this.win./*OK*/pageYOffset;
+    const host = this.ampdoc.getRootNode().host;
+    const hostOffset = (host && host.offsetTop) || 0;
+    return pageScrollTop - hostOffset;
```

Comment:
I think `getLayoutRect()` or `x.offsetTop` are both fine for the purposes of bubble - they are triggered by user action so we have more tolerance.

> Also, do the iOS embed binding classes also need to account for the offset of the shadow root inside the host page?

Don't know. You'd have to test on iOS.